CMakeLists.txt

```diff
@@ -14,90 +14,49 @@ project( vtu11 CXX )
-    if( CMAKE_COMPILER_IS_GNUCXX )
+# If compiler is g++: Enable further  warnings and treat all warnings as errors
+if( VTU11_STRICT_COMPILER AND CMAKE_COMPILER_IS_GNUCXX )
 
-        # Removed because of filesystem header:
-        # -Wpadded -Wswitch-default
+    set( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC -pedantic -Wall -Wextra -Wcast-align -Wsuggest-attribute=cold       \
+        -Wsuggest-attribute=pure -Wimport -Wsuggest-final-methods -Wsuggest-attribute=const -Wsuggest-attribute=format \
+        -Wsuggest-attribute=malloc -Wsuggest-attribute=noreturn -Wformat-y2k -Wpacked -Wno-restrict                    \
+        -Wswitch-enum -Wwrite-strings -Wformat-nonliteral -Wformat-security -Wcast-qual -Wsuggest-override             \
+        -Wsuggest-final-types -Wdisabled-optimization -Wformat=2 -Winit-self -Wlogical-op -Wmissing-include-dirs       \
+        -Wnoexcept -Wold-style-cast -Woverloaded-virtual -Wredundant-decls -Wshadow -Wsign-conversion -Wsign-promo     \
+        -Wstrict-null-sentinel -Wstrict-overflow=5 -Wundef -Werror" )
 
-        set( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC -pedantic -Wall -Wextra -Wcast-align -Wsuggest-attribute=cold       \
-            -Wsuggest-attribute=pure -Wimport -Wsuggest-final-methods -Wsuggest-attribute=const -Wsuggest-attribute=format \
-            -Wsuggest-attribute=malloc -Wsuggest-attribute=noreturn -Wformat-y2k -Wpacked -Wno-restrict                    \
-            -Wswitch-enum -Wwrite-strings -Wformat-nonliteral -Wformat-security -Wcast-qual -Wsuggest-override             \
-            -Wsuggest-final-types -Wdisabled-optimization -Wformat=2 -Winit-self -Wlogical-op -Wmissing-include-dirs       \
-            -Wnoexcept -Wold-style-cast -Woverloaded-virtual -Wredundant-decls -Wshadow -Wsign-conversion -Wsign-promo     \
-            -Wstrict-null-sentinel -Wstrict-overflow=5 -Wundef -Werror" )
+endif( VTU11_STRICT_COMPILER AND CMAKE_COMPILER_IS_GNUCXX )
 
-    endif( CMAKE_COMPILER_IS_GNUCXX )
+file( GLOB TEST_FILES test/*.cpp)
+file( GLOB_RECURSE INCLUDE_FILES vtu11/*.hpp)
 
-endif( VTU11_STRICT_COMPILER )
+add_executable( vtu11_testrunner ${TEST_FILES} ${INCLUDE_FILES} test/vtu11_testing.hpp )
 
-# -------------------- zlib dependency ---------------------------
+target_include_directories( vtu11_testrunner PRIVATE vtu11 )
 
 find_package( ZLIB )
 
 if( ZLIB_FOUND )
 
-    message( STATUS "Enabling vtu11 with zlib compression" )
+    message( STATUS "Enabling vtu11 tests with zlib compression" )
 
-    add_definitions( -DVTU11_ENABLE_ZLIB )
-    set( VTU11_ZLIB_LIBRARIES ${ZLIB_LIBRARIES} )
-
-    set( VTU11_ALL_INCLUDE_DIRS ${VTU11_ALL_INCLUDE_DIRS} ${ZLIB_INCLUDE_DIRS} )
+    target_link_libraries( vtu11_testrunner PRIVATE ${ZLIB_LIBRARIES} )
+    target_include_directories( vtu11_testrunner PRIVATE ${ZLIB_INCLUDE_DIRS} )
+    target_compile_definitions( vtu11_testrunner PRIVATE VTU11_ENABLE_ZLIB )
```

Comment:
we usually use `add_definitions` and I think Pooyan told me the reason a while back but I am fine either way :)

Reply:
Well I'm not sure abouit definitions, but the advice for link libraries and include directories is to use the target specific one. So I thought it might be similar for compile definitions.